[migration] TS client initial release release

supreme-gg-gg opened the pull request.

supreme-gg-gg commented:
The CI check for typescript isn't running yet but I believe it will start checking after I merged this it already exists in `prettier-ts.yml`. I will turn on the `xo` linter check for typescript in the action and the precommit hook once we fixed all the 600+ linter error LMAO

Quimisagi commented:
Oh, really looking forward to it 😆

endernoke commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,45 @@
+        files: '^(?!instagram-ts/).*\.py$'
+
+  # TypeScript/JavaScript tools (only for instagram-ts folder)
+  - repo: https://github.com/pre-commit/mirrors-prettier
```
I noticed that https://github.com/pre-commit/mirrors-prettier is archived.
Should we consider using https://github.com/rbubley/mirrors-prettier instead?

supreme-gg-gg commented:
> lgtm 😄 everything works!
> 
> Though there's a slight UI behavior difference between mock chat UI and the real one, i guess we can borrow some technical debt now lmao

Can you attach a screenshot of the issue?

endernoke commented:
Btw the `instagram-cli` package name is taken on npm (cli to show simple IG feed) but it is unmaintained. I think we could try contacting the owner to take over the name

endernoke commented:
> > lgtm 😄 everything works!
> > Though there's a slight UI behavior difference between mock chat UI and the real one, i guess we can borrow some technical debt now lmao
> 
> Can you attach a screenshot of the issue?

<img width="1920" height="1128" alt="image" src="https://github.com/user-attachments/assets/2e0ab4df-9fd4-4204-ba09-3bda82b197f5" />

supreme-gg-gg commented:
> > > lgtm 😄 everything works!
> 
> > > Though there's a slight UI behavior difference between mock chat UI and the real one, i guess we can borrow some technical debt now lmao
> 
> > 
> 
> > Can you attach a screenshot of the issue?
> 
> 
> 
> <img width="1920" height="1128" alt="image" src="https://github.com/user-attachments/assets/2e0ab4df-9fd4-4204-ba09-3bda82b197f5" />
> 
> 
> 
> Also it is not in alt screen is this intended

I forgot to put it in alr screen lol, but can you point out where is the difference between regular UI cuz it seems the same? The input box bugs but I think it's reported on a previous PR already and is fixable by resizing terminal

endernoke commented:
> > > > lgtm 😄 everything works!
> > 
> > 
> > > > Though there's a slight UI behavior difference between mock chat UI and the real one, i guess we can borrow some technical debt now lmao
> > 
> > 
> > > 
> > 
> > 
> > > Can you attach a screenshot of the issue?
> > 
> > 
> > <img alt="image" width="1920" height="1128" src="https://private-user-images.githubusercontent.com/184711003/486590790-2e0ab4df-9fd4-4204-ba09-3bda82b197f5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTczNDQ1OTgsIm5iZiI6MTc1NzM0NDI5OCwicGF0aCI6Ii8xODQ3MTEwMDMvNDg2NTkwNzkwLTJlMGFiNGRmLTlmZDQtNDIwNC1iYTA5LTNiZGE4MmIxOTdmNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwOFQxNTExMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NzdhYjM1ODRkN2M3NzBlY2JiOTRkZGEwOTQ5MzAxOGYxY2Y0YmU2N2ZiNGIxM2I1OTk5ZWRjYTJjYWI4OTBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.QwbcEOIR02v4eedZ0S9En4d44QPBWBEgfv50heaQxys">
> > Also it is not in alt screen is this intended
> 
> I forgot to put it in alr screen lol, but can you point out where is the difference between regular UI cuz it seems the same? The input box bugs but I think it's reported on a previous PR already and is fixable by resizing terminal

Notice that the bottom of the message history is clipped, I think I fixed it in the last PR by reversing the flexbox anchor but somehow the mock client is still having the bug. @supreme-gg-gg could u look into this if not I'll investigate tmr

supreme-gg-gg commented:
Added the base command when you run `npm run start, i.e. instagram` to be the same as python behaviour, help is shown after running `npm start -- --help, i.e. instagram --help`. Also added the types for `ascii-art` from here: https://www.npmjs.com/package/@types/ascii-art instead of using a custom-written one for better maintainability

endernoke commented:
> wait @endernoke I don't think this is reproducible 😭, also I think the mock chatUI is already wrapped in Alt screen 😄 if you check `source/cli.mock.tsx`

Let's pretend this doesn't exist then 😭 If technical creditors decide to chase us we'll figure out a fix.

endernoke commented:
> Added the base command when you run `npm run start, i.e. instagram` to be the same as python behaviour, help is shown after running `npm start -- --help, i.e. instagram --help`. Also added the types for `ascii-art` from here: https://www.npmjs.com/package/@types/ascii-art instead of using a custom-written one for better maintainability
> 
> This will be merged tomorrow, if there are any recurrent issues pls ping me

Nice!! Btw we might want to look into https://github.com/sindresorhus/ink-big-text as an alternative for rendering ansi text, since `ascii-art` has a `canvas` dependency which could be troublesome to install on certain platforms...

This PR has 100 commits now btw 😂

supreme-gg-gg commented:
lol we've broken new record!! Have u already removed ascii art if not I'll keep it for this PR and remove it next time cuz it might break stuff with ascii prolly isn't zero-shot lol

endernoke commented:
> lol we've broken new record!! Have u already removed ascii art if not I'll keep it for this PR and remove it next time cuz it might break stuff with ascii prolly isn't zero-shot lol

No remove it in future PRs lol let's not try to break things 10 mins before prod lmaoo

Quimisagi commented:
> > > > > Is it implemented on `feed`? I'm trying it on a PC different to the one I usually use so I don't know if it has something to do, but I tried feed and doesn't look like it's working fine.... The ASCII changes itself even if I don't press anything. Also takes tame to re-render when navigating through posts.
> > > > 
> > > > 
> > > > Hmmm... Tried my own branch and it's still wrong there. Huh, guess probably I didn't test it thoroughly last time, so probably there's still work to do.
> > > 
> > > 
> > > @Quimisagi wait try pull from `ts-migration/main` since we've pushed a lot of new changes with images / ascii. I have tried it again and it seems to be working properly, let me know if the issue persists or perhaps attach a screenshot
> > 
> > Tried this on the machine I usually use and its working fine! It's just a little bit off when I use it on my other laptop, I still have to figure out why...
> > 
> > I don't know if you can tell from the video, but sometimes despite I'm not moving the cursor the ASCII changes suddenly, or after moving the cursor it takes a while to render again. Also the index of the carousel is not displayed below despite it being a carousel post, nor I can open the media pressing 'o'.
> > 
> > As I mentioned before those problems doesn't happen on the machine I usually use, though.
> > 
> >  2025-09-11.00-29-28.mp4
> 
> this is very very interesting, this should not happen if you're running the exact same code on both machines... The carousel index issue might be caused by when we resolve the merge conflicts, but still the behaviour should be consistent across different install on different devices 😭 💀. Perhaps we can take this offline to the discord if this keeps happening and we can investigate more thoroughly than on here

Yeah
Will probably create a issue later (*´∀｀*)
